src/main/kotlin/project2/Network.kt

```diff
@@ -0,0 +1,57 @@
+package project2
+
+class Network(c:Int,trains:List<Train>, segmentcount:Int) {
+    private var capacity:Int=c
+    private var trains:List<Train> =trains
+    private var segments:MutableList<Segment> = mutableListOf()
+    private var segmentcount = segmentcount
+
+    public fun createSegments(){
```

Comment:
Da das unbedingt nach Erstellung eines `Network`-s durchgeführt werden muss, sollte es in einen initializer block (siehe [Dokumentation](https://kotlinlang.org/docs/reference/classes.html)):

```
init {
    // code
}
````